roles/config-quay-enterprise/tasks/complete_setup.yml

```diff
@@ -1,46 +1,53 @@
 ---
 
-- name: Set Setup Output File Location
-  set_fact:
-    tmp_setup_file_location: /tmp/quay
+- name: Configure Setup Process on 1st Quay Host
+  block:
+    - name: Set Setup Output File Location
+      set_fact:
+        tmp_setup_file_location: /tmp/quay
 
-- name: Flush Handlers
-  meta: flush_handlers
+    - name: Restart Quay
+      systemd:
+        name: "{{ quay_name }}"
+        enabled: yes
+        state: restarted
+        daemon_reload: yes
 
-- name: Hit Quay Setup Endpoint
-  uri:
-    url: "{{ quay_http_protocol }}://{{ quay_hostname }}/setup/"
-    validate_certs: no
-    method: GET
-    return_content: yes
-  register: uri_setup
-  until: uri_setup.status == 200
-  retries: 30
-  delay: 10
-- name: Write file
-  copy: 
-    content: "{{ uri_setup.content }}" 
-    dest: "{{ tmp_setup_file_location }}"
-- name: Extract csrf token
-  shell: cat /tmp/quay | grep "__token" | awk -F\' '{print $(NF-1)}'
-  register: token
-- name: Create Superuser
-  uri:
-    url:  "{{ quay_http_protocol }}://{{ quay_server_hostname }}/api/v1/superuser/config/createsuperuser?_csrf_token={{ token.stdout | urlencode }}"
-    validate_certs: no
-    method: POST
-    body_format: json
-    body:
-      username: "{{ quay_superuser_username }}"
-      email: "{{ quay_superuser_email }}"
-      password: "{{ quay_superuser_password }}"
-    headers:
-      Cookie: "{{ uri_setup.set_cookie }}"
+    - name: Hit Quay Setup Endpoint
+      uri:
+        url: "{{ quay_http_protocol }}://{{ quay_hostname }}/setup/"
+        validate_certs: no
+        method: GET
+        return_content: yes
+      register: uri_setup
+      until: uri_setup.status == 200
+      retries: 30
+      delay: 10
+    - name: Write file
+      copy: 
+        content: "{{ uri_setup.content }}" 
+        dest: "{{ tmp_setup_file_location }}"
+    - name: Extract csrf token
+      shell: cat /tmp/quay | grep "__token" | awk -F\' '{print $(NF-1)}'
+      register: token
+    - name: Create Superuser
+      uri:
+        url:  "{{ quay_http_protocol }}://{{ quay_server_hostname }}/api/v1/superuser/config/createsuperuser?_csrf_token={{ token.stdout | urlencode }}"
+        validate_certs: no
+        method: POST
+        body_format: json
+        body:
+          username: "{{ quay_superuser_username }}"
+          email: "{{ quay_superuser_email }}"
+          password: "{{ quay_superuser_password }}"
+        headers:
+          Cookie: "{{ uri_setup.set_cookie }}"
 
-- name: Delete Temporary Setup File
-  file:
-    state: absent
-    path: "{{ tmp_setup_file_location }}"
+    - name: Delete Temporary Setup File
+      file:
+        state: absent
+        path: "{{ tmp_setup_file_location }}"
+  when: inventory_hostname == groups['quay_enterprise'][0]
```

Comment:
Maybe a bit more of a "personal" preference, but I prefer the following syntax for "doing something on the first task" - seems a bit more elegant and potentially even less error prone in case group names changes, etc. 

`delegate_to: "{{ ansible_play_hosts | first }}"`

I'm using it here:
https://github.com/redhat-cop/infra-ansible/blob/v1.0.2/roles/config-dns-server/tasks/named_keys.yml#L14

Reply:
You can also include `run_once: true` just to ensure it doesn't run more than once, but it is redundant.

Reply:
@oybed I use this convention throughout the quay ecosystem of roles. I would recommend a cleanup after the merge that can address all concerns

Reply:
Sounds good - it was just a recommendation